src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -1487,13 +1509,20 @@ def forward(self, inputs, targets):
                     keep_model = self.net
                     self.net = self.ema_model.ema
 
+                train_inf_time = timer.stop()
+                self._write_scalars_to_logger(
+                    metrics=train_metrics_dict, epoch_values=self._epoch_start_logging_values, epoch=1 + epoch, inference_time=train_inf_time, tag="Train"
+                )
+
                 # RUN TEST ON VALIDATION SET EVERY self.run_validation_freq EPOCHS
                 valid_metrics_dict = {}
-                if (epoch + 1) % self.run_validation_freq == 0:
+                is_validation_calculated = self._should_run_validation_for_epoch(epoch)
+
+                if is_validation_calculated:
```

Comment:
I think it's safe to inline the call as follows:

```
if self._should_run_validation_for_epoch(epoch):
   ...
```

IMO it's a bit clearer and does not introduce temp variable `is_validation_calculated`

Reply:
I assumed we need to check this condition several times, so we can store the result in a variable and not call the method every time. This way the code looks more compact and, from my point of view, does not harm the readability.